modules/siddhi-core/src/main/java/org/wso2/siddhi/core/query/input/stream/state/LogicalPreStateProcessor.java

```diff
@@ -82,13 +95,21 @@ public void setStartState(boolean isStartState) {
 
     @Override
     public void resetState() {
-        pendingStateEventList.clear();
-        partnerStatePreProcessor.pendingStateEventList.clear();
-
-        if (isStartState && newAndEveryStateEventList.isEmpty()) {
-            //        if (isStartState && stateType == StateInputStream.Type.SEQUENCE && newAndEveryStateEventList
-            // .isEmpty()) {
-            init();
+        if (logicalType == LogicalStateElement.Type.OR || pendingStateEventList.size() ==
+                partnerStatePreProcessor.pendingStateEventList.size()) {
+            pendingStateEventList.clear();
+            partnerStatePreProcessor.pendingStateEventList.clear();
+
+            if (isStartState && newAndEveryStateEventList.isEmpty()) {
+                //        if (isStartState && stateType == StateInputStream.Type.SEQUENCE && newAndEveryStateEventList
```

Comment:
Better remove this line.

Reply:
Hi,
Thanks for pointing this. Please merger the PR https://github.com/wso2/siddhi/pull/566

Thanks